lib/pbench/server/__init__.py

```diff
@@ -64,6 +65,109 @@ def tstos(ts: float = None) -> str:
+    """
+    return tstos(_time())
+
+
+def _get_valid_path(
+    env_name: str, dir_val: str, logger: Optional[Logger]
+) -> Optional[Path]:
+    """Get the fully resolved path from the given path.
+
+    If a logger is given, will emit error logs when the directory is not
+    found or the resolved name is not a directory.
+
+    Args:
+        env_name : Legacy environment name to display in error messages
+        dir_val : A directory value to resolve to a real directory
+        logger : Logger to use when low-level error messages should be
+            emitted
+
+    Returns:
+        A Path directory object of the directory on disk, None otherwise.
+    """
+    try:
+        dir_path = Path(dir_val).resolve(strict=True)
+    except FileNotFoundError:
+        if logger:
+            logger.error(
+                "The {} directory, '{}', does not resolve to a real location",
+                env_name,
+                dir_val,
+            )
+        return None
+    else:
+        if not dir_path.is_dir():
+            if logger:
+                logger.error(
+                    "The {} directory, does not resolve to a directory ('{}')",
+                    env_name,
+                    dir_path,
+                )
+            return None
+    return dir_path
+
+
+def get_resolved_dir(
+    env_name: str, dir_val: str, logger: Optional[Logger]
+) -> Optional[Path]:
+    """Get the resolved directory associated with the environment name.
+
+    Args:
+        env_name : Legacy environment name to display in error messages
+        dir_val : A directory value to resolve to a real directory
+        logger : Logger to use when low-level error messages should be
+            emitted
+
+    Returns:
+        A Path directory object, None if the directory value does not
+            resolve to a real directory.
+    """
+    dir_path = _get_valid_path(env_name, dir_val, logger)
+    if not dir_path:
+        return None
+
+    return dir_path
+
+
+def _get_valid_dir_option(
+    conf: ConfigParser, env_name: str, section: str, option: str
+) -> Path:
+    """Get the validated directory option from the given section.
+
+    Args:
+        conf : The ConfigParser object to use for looking up the option
+        env_name : Legacy environment name to display in error messages
+        section : The configuration section name to find the option
+        option : The option name to find in the given section
+
+    Raises:
+        BadConfig : if the directory option does not resolve to a directory
+            on the file system, or if the option is missing, or if the
+            section is missing
+
+    Returns:
+        A Path directory object.
+    """
+    try:
+        dir_val = conf.get(section, option)
+    except (NoOptionError, NoSectionError) as exc:
+        raise BadConfig(str(exc))
+    else:
+        if not dir_val:
+            raise BadConfig(f"option {option} in section {section} is empty")
```

Comment:
As I [said](https://github.com/distributed-system-analysis/pbench/pull/3170#discussion_r1081596245) in #3170....

The `else` and indented `if` look funny to me.  I would omit the `else`.

Also, I would offer the following alternative instead:
```
        try:
            dir_val = self.conf.get(section, option)
            if not dir_val:
                raise ValueError(f"option {option} in section {section} is empty")
        except (NoOptionError, NoSectionError, ValueError) as exc:
            raise BadConfig(str(exc))
```
This reports all errors via `BadConfig` from a single, common `raise` statement while still capturing the nature of the various triggering failures.